card/spec/support/matchers/card_matchers.rb

```diff
@@ -40,21 +40,45 @@ def expect_card *marks
     match do |card|
       values_match?(name, card.send(field))
     end
+
+    failure_message do |card|
+      super() + ", but was #{card.send(field)}"
+    end
   end
 end
 
-# RSpec::Matchers.define :have_codename do |codename|
-#   match do |card|
-#     values_match?(codename, card.codename)
-#   end
-# end
-#
-# RSpec::Matchers.define :have_content do |content|
-#   match do |card|
-#     puts card
-#     values_match?(content, card.content)
-#   end
-# end
+RSpec::Matchers.define :have_a_field do |field_key|
+  chain(:with_content) do |content|
+    @content = content
+  end
+
+  chain(:pointing_to) do |pointing_to|
+    @pointing_to = pointing_to
+  end
+
+  match do |card|
+    return unless card.is_a?(Card)
+    return unless (@field = card.fetch(trait: field_key))
+
+    if @content
+      values_match?(@content, @field.content)
+    elsif @pointing_to
+      values_match?(:pointer, @field.type_code) &&
+        values_match?(@field.content, /\[\[#{@pointing_to}\]\]/)
+    else
+      values_match(Card, @field.class)
+    end
+  end
+
+  failure_message do |card|
+    return super(card) unless @field
+    if @content
+      "expected #{card} to have a field '#{field_key}' with content '#{@content}',
+but content is #{card.content.present? ? "empty" : card.content}"
+    elsif @pointing_to
+      "expected #{card} to have a field #{field_key} pointing to #{@pointing_to} but
+content is #{card.content}"
+    end
```

Comment:
I think this is missing an `end`, which is why rspecs are failing.

Reply:
no, two `end`s :simple_smile: